tests/tasks/setup_candlepin.yml

```
@@ -37,23 +64,22 @@
                  ternary('ansible.posix.rhel_rpm_ostree', omit) }}"
 
     - name: Stop and remove Candlepin container
-      containers.podman.podman_container:
-        name: candlepin
-        state: absent
+      command:
+        argv:
+          - podman
+          - stop
+          - candlepin
+      register: podman_stop_status
+      failed_when:
+        - podman_stop_status.rc != 0
+        - '"no such container" not in podman_stop_status.stderr'
+      changed_when: false
 
     - name: Start Candlepin container
```

```suggestion
    - name: Start Candlepin 
      vars:
        podman_run_cli:
          - podman
          - run
          - --rm
          - --detach
          - --hostname
          - "{{ lsr_rhc_test_data.candlepin_host }}"
          - --name
          - candlepin
          - --publish
          - 8443:8443
          - --publish
          - 8080:8080
          - "{{
            '--privileged'
            if ansible_distribution in ['CentOS', 'RedHat'] and ansible_distribution_major_version | int < 8
            else '' }}"
          - ghcr.io/ptoscano/candlepin-unofficial    
```

Instead of setting three facts, let's set the cli here

---

oh wow, is this syntax actually working? let's see...

---

this doesn't work: in case `--priviledged` is not set, the result is an empty item in the list, and this empty item becomes an empty parameter to podman, which then fails:
```
STDERR:

Error: parsing reference "": repository name must have at least one component

```

---

Oh, okay. One more thing you can try is using `omit`, i.e. replace `else '' }}"` with `else omit }}"`.
And if this doesn't work too, you can do what you did with set_fact but with `vars`, like this:

```
podman_run_cli: "{{
  podman_run_cli + ['--privileged']
  if ansible_distribution in ['CentOS', 'RedHat'] and ansible_distribution_major_version | int < 8
  else podman_run_cli }}"
podman_run_cli: "{{ podman_run_cli + ['ghcr.io/ptoscano/candlepin-unofficial'] }}"
```

---

ah, a bit of jinja filtering will do the job -- thanks for the hint!

---

set_fact is not welcomed because it creates an Ansible fact that is kept after playbook finished, we want to avoid this if we can.